File: models/product.go
package models

import "gorm.io/gorm"

type Product struct {
	gorm.Model
	Name        string         `gorm:"size:100;not null"`
	Slug        string         `gorm:"size:100;uniqueIndex;not null"`
	Description string         `gorm:"type:text"`
	Price       int64          `gorm:"not null;index"`
	Stock       int            `gorm:"not null;default:0"`
	IsActive    bool           `gorm:"not null;default:true"`
	CategoryID  uint           `gorm:"index"`
	Category    Category       `gorm:"constraint:OnUpdate:CASCADE,OnDelete:RESTRICT;"`
	Images      []ProductImage `gorm:"constraint:OnUpdate:CASCADE,OnDelete:CASCADE;"`

	Tastes []ProductTaste `gorm:"constraint:OnUpdate:CASCADE,OnDelete:CASCADE;"`
}

type ProductTaste struct {
	gorm.Model
	ProductID uint   `gorm:"index;not null"`
	Name      string `gorm:"size:100;not null"`
}

File: frontend/src/types/product.ts
// src/types/product.ts

export interface ProductImage {
    id: number;
    url: string;
    is_primary: boolean;
    sort_order: number;
}

// Ответ бэка: ProductResponse (Go) отдаёт tastes как []string,
// поэтому на фронте просто массив строк.
export interface Product {
    id: number;
    name: string;
    slug: string;
    description: string;
    price: number;
    stock: number;
    is_active: boolean;
    category_id: number;
    images: ProductImage[];
    tastes: string[]; // <-- НОВОЕ ПОЛЕ
}

export interface ProductsListResponse {
    ok: boolean;
    data: {
        page: number;
        limit: number;
        total: number;
        items: Product[];
    };
}

// Если хочешь типизировать одиночный продукт:
export interface ProductResponse {
    ok: boolean;
    data: Product;
}
